stustapay/tse/diebold_nixdorf_usb/simulator.py

```diff
@@ -278,6 +290,62 @@ def unblockuser(self, msg):
         response["Status"] = "ok"
         return response
 
+    def registerclientid(self, msg):
+        response = {}
+        # check if all Parameters are here
+        if "ClientID" not in msg or "Password" not in msg:
+            return dnerror(3)  # param missing
+        # check password_admin == 12345
+        try:
+            if base64.b64decode(msg["Password"]).decode("utf8") != self.password_admin:
+                self.password_block_counter += 1
+                return dnerror(30)  # password wrong
+        except:
+            return dnerror(9)  # base64
+
+        if len(msg["ClientID"]) > 30:
+            return dnerror(4)
+
+        if len(self.current_transactions) > 15:
+            return dnerror(23)
+
+        if msg["ClientID"] in self.current_transactions:
+            return {"Status": "error", "Code": 1337, "Desc": "Client already registered"}
+
+        self.current_transactions[msg["ClientID"]] = set()
+
+        self.password_block_counter = 0
+        response["Status"] = "ok"
+        return response
+
+    def deregisterclientid(self, msg):
+        response = {}
+        # check if all Parameters are here
+        if "ClientID" not in msg or "Password" not in msg:
+            return dnerror(3)  # param missing
+        # check password_admin == 12345
+        try:
+            if base64.b64decode(msg["Password"]).decode("utf8") != self.password_admin:
+                self.password_block_counter += 1
+                return dnerror(30)  # password wrong
+        except:
```

Comment:
same here, should not be a catch-all, instead just the base64 decoding error

Reply:
now binascii.error